src/map.rs

```diff
@@ -4380,6 +4398,24 @@ impl<'a, 'b, K, Q: ?Sized, V, S, A: Allocator> VacantEntryRef<'a, 'b, K, Q, V, S
         &mut entry.1
     }
 
+    /// provide explicit key at insert-time instead of relying on there being effectively a from &K to K implementation and not working with cloneable values
+    #[cfg_attr(feature = "inline-more", inline)]
+    pub fn insert_kv(self, key: K, value: V) -> &'map mut V
```

Comment:
I feel that a better name would be `insert_with_key`. This indicates that the key is being explicitly provided.

Reply:
done